### PR TITLE
Typing notification event from peer

### PR DESCRIPTION
**Pull request by bitfriend**

### CODE REVIEW

**Comment by bitfriend** [comment]
Now we can get event receiver using `SyncState::getEphemeralEventRx` on flutter side

File: app/lib/common/widget/CrossSigning.dart [hunk 1]
```diff
@@ -5,7 +5,7 @@ import 'dart:async';
 import 'package:effektio/common/store/separatedThemes.dart';
 import 'package:effektio/common/widget/AppCommon.dart';
 import 'package:effektio_flutter_sdk/effektio_flutter_sdk_ffi.dart'
-    show SyncState, CrossSigningEvent, Client, FfiListEmojiUnit;
+    show Client, FfiListEmojiUnit, GenericEvent, SyncState;
```
Comment (gtalha07) [inline app/lib/common/widget/CrossSigning.dart]: Using the different class here. Does this handle both emoji and other events? 
Note: Just for my reminder, the PR push can break the implemented cross-signing UI and I need to fix it.

**Comment by gtalha07** [comment]
> @bitfriend I am confused. this seems to contain a lot of unrelated changes. what does the emoji verification have to do with typing notification? please clarify...

Seems like the CrossSigningEvent handler is changed into GenericEvent handler, which has support of listening both cross signing and invitation events?. Please correct me if I'm wrong @bitfriend

**Comment by bitfriend** [comment]
@gtalha07 I guess it is better to change `startCrossSigning` to `install`.
Later we may have to implement `uninstall` function too.

File: native/cli/Cargo.toml [hunk 2]
```diff
@@ -21,6 +21,8 @@ term-table = "1.3.2"
 
 [dependencies.effektio-core]
 path = "../core"
+default-features = false
```
Comment (gnunicorn) [inline native/cli/Cargo.toml:24]: Are you sure we want these?

File: native/effektio-test/src/lib.rs [hunk 3]
```diff
@@ -1,6 +1,8 @@
 #[cfg(test)]
 mod auth;
-#[cfg(test)]
-mod emoji_verification;
+// #[cfg(test)]
```
Comment (gnunicorn) [inline native/effektio-test/src/lib.rs:3]: What's the point of this? Doesn't the test work anymore? Can we please leave is enabled led?

**Review by gnunicorn — APPROVED** [review]
Let's fix my small comments a d merge this. But let's rethink that giant sync-callback. I think we probably should switch to type-specific event-handlers instead (after this PR).

File: native/cli/Cargo.toml (hunk 2 above)
Comment (bitfriend) [reply native/cli/Cargo.toml:24]: image
I inserted that code to reset database.
I think anyone would need mock data when he wants to reset his local synapse server.
What do you think?

File: native/effektio-test/src/lib.rs (hunk 3 above)
Comment (bitfriend) [reply native/effektio-test/src/lib.rs:3]: As I said, when `emoji_verification` test is enabled, `typing_notification` test is not working.
I guess frequent usage of registered accounts of synapse server has some problem.
When performing `typing_notification` test, should I make new account to synapse server?

Comment (gnunicorn) [reply native/effektio-test/src/lib.rs:3]: That's weird. Let's merge as is, but file a bug - this needs fixing, synapse should be able to deal with both.

File: native/cli/Cargo.toml (hunk 2 above)
Comment (gnunicorn) [reply native/cli/Cargo.toml:24]: I don't understand. Why did you disabled default features on this dependency?

Comment (bitfriend) [reply native/cli/Cargo.toml]: Fixed by https://github.com/effektio/effektio/pull/170/commits/5cad8afaf03ceaadbf8e76c35ab1e5c0257cbc73

File: native/effektio-test/src/lib.rs (hunk 3 above)
Comment (bitfriend) [reply native/effektio-test/src/lib.rs:3]: Just opened #202 and assigned it to you and me, because it is rust bug.

**Pull request merged**